Contains on embedded data types may be broken, and it's unclear how it's supposed to work

There don't seem to be any tests that use the .contains method on an embedded data property (EmbeddedSet etc.) and I can't figure out how to use it without getting weird errors. Consider the following code:
```python
BaseNode = declarative_node()

class Person(BaseNode):
    element_type = 'Person'
    element_plural = 'Person'

    name = String(nullable=False)
    aliases = EmbeddedSet(nullable=False)

class ModelsTest(unittest.TestCase):
    def setUp(self):
        self.graph = get_test_orientdb_graph()
        self.graph.create_all(BaseNode.registry)

    def test_get_person_by_name(self):
        person1 = self.graph.Person.create(name='Hello World',
                                           aliases=set(['Hello T. World', 'World']))
        name = 'Hello World'
        result = self.graph.query(Person).filter(
            (Person.name == name) | Person.aliases.contains(name)).all()
        self.assertSetEqual(set(result), set([person1]))
```

This fails with the following trace:
```
.../pyorient/ogm/query.py:145: in prepare
    wheres = rid_clause + self.build_wheres(params)
.../pyorient/ogm/query.py:411: in build_wheres
    exp_where = [self.filter_string(filter_exp)] if filter_exp else []
.../pyorient/ogm/query.py:339: in filter_string
    , self.filter_string(right))
.../pyorient/ogm/query.py:320: in filter_string
    left_str, self.filter_string(right))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <pyorient.ogm.query.Query object>, expression_root = 'Hello World'

    def filter_string(self, expression_root):
>       op = expression_root.operator
E       AttributeError: 'str' object has no attribute 'operator'
```

Specifically, [this line](https://github.com/mogui/pyorient/blob/master/pyorient/ogm/query.py#L323) seems to be the problem, and I don't understand why it recursively calls `filter_string` on the right-hand side. Can the RHS be an expression? I would have expected it to always be a variable, because I don't see how one of these data structures could be tested for containing the result of an expression.

--- a/pyorient/ogm/query.py
+++ b/pyorient/ogm/query.py
@@ -1,4 +1,5 @@
-from .operators import Operator, RelativeOperand, Operand, ArithmeticOperation
+from .operators import (Operator, RelativeOperand, Operand,
+                        ArithmeticOperation, LogicalConnective)
 from .property import Property, PropertyEncoder
 from .element import GraphElement
 from .exceptions import MultipleResultsFound, NoResultFound
@@ -316,11 +317,15 @@ class Query(object):
                                                       , right, self))
             elif op is Operator.Between:
                 far_right = PropertyEncoder.encode(expression_root.operands[2])
-                return '{0} BETWEEN {1} and {2}'.format(
+                return u'{0} BETWEEN {1} and {2}'.format(
                     left_str, right, far_right)
             elif op is Operator.Contains:
-                return u'{0} contains({1})'.format(
-                    left_str, self.filter_string(right))
+                if isinstance(right, LogicalConnective):
+                    return u'{0} contains({1})'.format(
+                        left_str, self.filter_string(right))
+                else:
+                    return u'{} in {}'.format(
+                        PropertyEncoder.encode(right), left_str)
             elif op is Operator.EndsWith:
                 return u'{0} like \'%{1}\''.format(left_str, right)
             elif op is Operator.Is:

--- a/tests/test_ogm.py
+++ b/tests/test_ogm.py
@@ -475,6 +475,21 @@ class OGMEmbeddedTestCase(unittest.TestCase):
         self.assertDictContainsSubset(result.children, children)
         self.assertDictContainsSubset(children, result.children)
 
+    def testEmbeddedSetContains(self):
+        g = self.g
+
+        name = 'embed'
+        alias = ['implant', 'lodge', 'place']
+        element_cls = g.registry['emb_set']
+
+        g.emb_set.create(name=name, alias=alias)
+        canonical_result = g.emb_set.query(name=name).one()
+        self.assertIsNotNone(canonical_result)
+
+        for alternate in alias:
+            received = g.query(element_cls).filter(element_cls.alias.contains(alternate)).one()
+            self.assertEqual(canonical_result, received)
+
 
 if sys.version_info[0] < 3:
     def to_unicode(x):

